lib/Interpreter/Transaction.cpp

```diff
@@ -68,7 +68,12 @@ namespace cling {
           if (name.equals(ND->getNameAsString()))
             return ND;
         }
-        else if (LinkageSpecDecl* LSD = dyn_cast<LinkageSpecDecl>(DI)) {
+      }
+    }
+    // Not found yet, peek inside extern "C" decalrations
+    for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
```

Comment:
Yes, and no....
```
[cling]$ f(4)
input_line_6:2:2: error: use of undeclared identifier 'f'
 f(4)
 ^
[cling]$ namespace f {}
input_line_7:1:11: error: redefinition of 'f' as different kind of symbol
namespace f {}
          ^
```
If you're specifically trying to limit per-transaction, you'll have to give up some corner cases that won't work or a lot of convolution matching the spec.
And you'll have to match clang's interpretation of the spec as clang will be the ultimate decider, so why not just let it do its thing?